frontend/src/metabase/components/ExplicitSize/ExplicitSize.tsx

```diff
@@ -208,10 +208,15 @@ function ExplicitSize<T>({
             return;
           }
 
+          // Use requestAnimationFrame to avoid layout thrashing.
+          // This fixes the "ResizeObserver loop completed with undelivered notifications" error.
+          // Refer to https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver#observation_errors
           if (this.state.width !== width || this.state.height !== height) {
-            this.setState({ width, height }, () =>
-              this.props?.onUpdateSize?.(),
-            );
+            requestAnimationFrame(() => {
```

Comment:
@heypoom I noticed that mantine saves the id of the request in order to be able to clear it with cancelAnimationFrame, I'm wondering if not doing it could lead to the react warning/error of calling setState on an unmounted component

Reply:
@npretto good catch! let me add that in.

Reply:
fixed in https://github.com/metabase/metabase/pull/47606/commits/5f0e9e364e8625af5f2b6e77976830580a56c6cd, thanks!